crates/primitives/gateway/src/receipt.rs

```diff
@@ -146,12 +146,24 @@ pub struct ExecutionResources {
     pub builtin_instance_counter: BuiltinCounters,
     pub n_steps: u64,
     pub n_memory_holes: u64,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub data_availability: Option<L1Gas>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
This is not a problem with bincode?

Reply:
Since the gateway primitives are never stored on disk but only used for data serialization to and from fgw responses, this never ends up interacting with bincode and is therefore safe. I am a bit worried though at how easy it is for us to call bincode on unsupported data formats at the moment, and if it might not be wiser to switch to newer versions of bincode which do not have this ambiguity with serde.